[Felicia Ivane Pranoto] iP

feliciaivane opened the pull request:
# Bob

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Bob will help you keep track of all the tasks you have to do! It is
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use
* entertaining

How can you use Bob on your own computer?
1. download the file
2. double-click the file
3. add your tasks
4. sit back and let Bob do the rest! :grin:

And the best part is, Bob is completely **FREE**!

Bob's features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

You can use Bob to improve your Java skills too! Here's Bob's `<main>` method:

```java
public static void main(String[] args) {
    String currDirectory = new File("").getAbsolutePath();
    String dataDirectory = currDirectory + "/data";
    new Bob(dataDirectory).run();
}
```

Commit:
Add Gradle support

feliciaivane committed:
Complete Level 1, rename Duke to Bob

feliciaivane committed:
Complete level 2

feliciaivane committed:
Complete Level 3

feliciaivane committed:
Complete Level 4

feliciaivane committed:
Complete A-TextUiTesting

feliciaivane committed:
Complete Level 5

feliciaivane committed:
Complete Level 6

feliciaivane committed:
Complete Level 7

Support saving of tasks to hard disk when task list changes
and loading of tasks from hard disk when Bob starts up.

feliciaivane committed:
Complete Level 8

Teach Bob how to understand deadline and event
dates as actual Java LocalDate instead of String

feliciaivane committed:
Handle exceptions in Level 7 implementation

Catch and handle exceptions instead of just throwing them.

feliciaivane committed:
Revert "Handle exceptions in Level 7 implementation"

This reverts commit b841dacf31992645562a6853ac92204561e3d098.

feliciaivane committed:
Merge branch 'branch-Level-8'

Resolve conflicts in Event and Deadline class
and fix initialiser method in TaskList class.

feliciaivane committed:
Fix minor errors from previous commit

feliciaivane committed:
Merge branch 'branch-Level-7'

Fix previous mistake of merging Level 7 branch
incorrectly.

feliciaivane committed:
Add Ui, Storage and Parser class

Increase use of OOP in code.

feliciaivane committed:
Commit test files

feliciaivane committed:
Move all classes into bob package

feliciaivane committed:
Add Junit tests for Task and Todo class

feliciaivane committed:
Package app as JAR file

Complete A-Jar and make minor change to Ui class.

feliciaivane committed:
Add JavaDoc comments

Add JavaDoc comments to all non-private classes/methods,
and non-trivial private methods.

feliciaivane committed:
Add JavaDoc comments

Add missing JavaDoc comments.

feliciaivane committed:
Comply with coding standard

feliciaivane committed:
Add find feature

Allow users to search for tasks using a specific keyword.

feliciaivane committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments
  Add JavaDoc comments

feliciaivane committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Comply with coding standard

# Conflicts:
#	src/main/java/bob/Parser.java
#	src/main/java/bob/TaskList.java
#	src/main/java/bob/task/Event.java
#	src/main/java/bob/task/Todo.java

feliciaivane committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find feature

# Conflicts:
#	src/main/java/bob/Parser.java
#	src/main/java/bob/Ui.java

feliciaivane committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

feliciaivane committed:
Set up Gradle

feliciaivane committed:
Add GUI

GUI does not correspond to actual Bob yet.
GUI only returns user input back to them.

feliciaivane committed:
Make GUI functional

GUI now corresponds to Bob.
GUI shows Bob's actual responses to the user input.

feliciaivane committed:
Make trivial change for A-Gradle branch

feliciaivane committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Make trivial change for A-Gradle branch

feliciaivane committed:
Undo trivial change for Level-10 branch

feliciaivane committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Undo trivial change for Level-10 branch

markuslim24 commented on build.gradle:
```diff
@@ -0,0 +1,62 @@
+}
+
+application {
+    mainClassName = "bob.Bob"
```
Remember to change mainClassName to Launcher so the build uses Launcher as the entry point :)

markuslim24 commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,209 @@
+
+    /** Image representing Bob in the chat */
+    private Image bob = new Image(this.getClass().getResourceAsStream("/images/DaBob.jpeg"));
+
```
Wow, great that you described all your variables, definitely improves readability!

markuslim24 commented on src/main/java/bob/Storage.java:
```diff
@@ -0,0 +1,192 @@
+     *
+     * @param taskList New list of tasks to be saved to the bob.txt file.
+     */
+    public void updateBobFile(TaskList taskList){
```
May have missed out a whitespace here behind the parenthesis

markuslim24 commented on src/main/java/bob/exception/InvalidDateException.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Represents the exception thrown by Bob when the user inputs an invalid date as a deadline or event timing.
+ */
+public class InvalidDateException extends BobException{
```
Oops another whitespace here behind the parenthesis is also missing

markuslim24 reviewed (commented):
Overall a well documented codebase that pretty much follows the java coding standard, other than the very minor missing whitespaces here and there. The javadoc comments are really clear and its easy to understand what the classes/methods do based on the description!

feliciaivane committed:
Add assertions

No assertions are present.
Important assumptions are not being documented.

Defining assertions helps to detect bugs in the code since the runtime
can verify the assumptions.

Let's,
* add assertions to the code

feliciaivane committed:
Improve code quality

Code quality has not been critically examined.

It is important to ensure the written code is of a high quality.
This means the code should be readable with the appropriate comments,
and unsafe practices should be avoided.

Let's,
* Critically examine all the written code following the topics learnt
in the module, and ensure all written code adheres to the code
quality guidelines.

feliciaivane committed:
Merge pull request #1 from feliciaivane/branch-A-Assertions

Add assertions

feliciaivane committed:
Merge branch 'master' of github.com:feliciaivane/ip

* 'master' of github.com:feliciaivane/ip:
  Add assertions

feliciaivane committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions

# Conflicts:
#	src/main/java/bob/Parser.java
#	src/main/java/bob/Storage.java
#	src/main/java/bob/TaskList.java

feliciaivane committed:
Merge pull request #2 from feliciaivane/branch-A-CodeQuality

Branch a code quality

feliciaivane committed:
Merge branch 'master' of github.com:feliciaivane/ip

* 'master' of github.com:feliciaivane/ip:
  Improve code quality

feliciaivane committed:
Provide in-App guidance to users

Add help command that gives users link to help page.

feliciaivane committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
  Provide in-App guidance to users

feliciaivane committed:
Create gradle.yml

feliciaivane committed:
Set up checkstyle

Use checkStyle to detect coding style violations.

feliciaivane committed:
Improve GUI

feliciaivane committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Improve GUI

feliciaivane committed:
Set theme jekyll-theme-modernist

feliciaivane committed:
Create user guide

feliciaivane committed:
Fix deadline and event command bug

feliciaivane committed:
Edit user guide

feliciaivane committed:
Add representative screenshot

feliciaivane committed:
Set theme jekyll-theme-midnight

feliciaivane committed:
Fix checkstyle issues

feliciaivane committed:
Merge commit '1bdce2cce73095eb06a2d60f75160cc858a1db4e'

* commit '1bdce2cce73095eb06a2d60f75160cc858a1db4e':
  Set theme jekyll-theme-midnight

feliciaivane committed:
Change Ui.png

feliciaivane committed:
Edit both READMEs

feliciaivane committed:
Edit both READMEs

feliciaivane committed:
Fix README formatting issues

feliciaivane committed:
Fix User Guide formatting issues

feliciaivane committed:
Add Ui.png to READMEs

feliciaivane committed:
Update README

feliciaivane committed:
Add abstractions in Storage and TaskList classes

feliciaivane committed:
Add more abstraction to Parser class

feliciaivane committed:
Add more abstraction to Bob class

feliciaivane committed:
Separate JavaFx from Bob class

feliciaivane committed:
Fix use of exceptions

feliciaivane committed:
Fix checkstyle

feliciaivane committed:
Split Parser into two classes

feliciaivane committed:
Create parser package

feliciaivane committed:
Fix hyperlink for non-macbooks
